### PR TITLE
AVS 1.0.0-preview.1 SDK

### PR DESCRIPTION
**Pull request by ctaggart**
The is the beginning of a .NET SDK for the Microsoft.AVS service. I used a local swagger file that is in this pull request https://github.com/Azure/azure-rest-api-specs/pull/9773 . It currently just contains the generated code. I haven't made any [tests using the test framework](https://github.com/Azure/adx-documentation-pr/blob/master/engineering/adx_netsdk_process.md#writing-tests).

### CODE REVIEW

**Comment by isra-fel** [comment]
Hi @ctaggart according to our guideline if you want to check in your code to `master` branch, the SDK must be generated from the master branch of Azure/azure-rest-api-spc , and it must be covered by tests.

**Comment by ctaggart** [comment]
Should I change the namespace from `Microsoft.Azure.Management.AVS` to `Microsoft.Azure.Management.Avs`? Looking around the projects, it looks like I should.

Visual Studio freezes every time I try to open up either project I created and I'm not sure why. Can someone else try?

Is there something in the test base that can create and teardown a test resource group? I ask, because that is what the azdev python test API does. It is really convenient.

**Comment by isra-fel** [comment]
> Should I change the namespace from Microsoft.Azure.Management.AVS to Microsoft.Azure.Management.Avs? Looking around the projects, it looks like I should.

Yes

> Visual Studio freezes every time I try to open up either project I created and I'm not sure why. Can someone else try?

Happens to me too. Not sure why. VSCode works.

> Is there something in the test base that can create and teardown a test resource group? I ask, because that is what the azdev python test API does. It is really convenient.

Common practice is to have your test project reference resource manager SDK `Microsoft.Azure.Management.ResourceManager` and use its client to create resource groups. E.g. https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/appplatform/Microsoft.Azure.Management.AppPlatform/tests/Tests/SpringCloudTests.cs#L34

**Comment by isra-fel** [comment]
I just added a commit to enable CI for this new module

File: sdk/avs/Microsoft.Azure.Management.Avs/src/Microsoft.Azure.Management.Avs.csproj [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <!-- Please do not move/edit code below this line -->
+  <Import Project="$(RepoEngPath)/mgmt/AzSdk.reference.props" />
+  <!-- Please do not move/edit code below this line -->
+
+  <PropertyGroup>
+    <PackageId>Microsoft.Azure.Management.Avs</PackageId>
+    <Description>Provides developers with libraries for Azure VMware Solution.</Description>
+    <Version>0.1.0</Version>
```
Comment (isra-fel) [inline sdk/avs/Microsoft.Azure.Management.Avs/src/Microsoft.Azure.Management.Avs.csproj:9]:
```suggestion
    <Version>1.0.0-preview.1</Version>
```
The first version of a new module is typically `1.0.0-preview.1` according to our versioning guide: https://github.com/Azure/azure-sdk-for-net/blob/master/doc/dev/Versioning.md
Then it goes to 1.0.0-preview.2 and finally 1.0.0

**Review by isra-fel — COMMENTED** [review]
Another thing is, you may have noticed that our CI failed due to timeout. I have a commit checked in to master to solve the problem, so please merge or rebase master to get the fix. Thanks

**Comment by ctaggart** [comment]
Thanks @isra-fel. The initial test now passes. My goal is to add test coverage that matches the [az vmware tests](https://github.com/Azure/az-vmware-cli/blob/8251f6ca3d0eec31028c606456dc1ea890bd5300/azext_vmware/tests/latest/test_vmware_scenario.py#L84-L156).

**Comment by ctaggart** [comment]
Hi @isra-fel, please review and merge if approved. Basic tests have been added for major functionality. I'll add more with the next API version coming up soon.

**Review by isra-fel — APPROVED** [review]
Looks good to me!

**Comment by ctaggart** [comment]
Thanks for merging! How do I get this built and in the dev feed? I don't see `Microsoft.Azure.Management.Avs` in:
https://dev.azure.com/azure-sdk/public/_packaging?feed=azure-sdk-for-net

**Comment by isra-fel** [comment]
We have [a guide here](https://github.com/Azure/adx-documentation-pr/blob/master/engineering/adx_netsdk_process.md#signing-and-publishing-nuget-packges) to build and publish your module to nuget.org, which is basically every dotnet SDK is doing.

I'm not familiar with dev feed, but if you are sure that's where you want your module to go, I think you can follow the same process but choose not to publish, and upload the bit to dev feed manually.

**Pull request merged**